### PR TITLE
fix: allow searching in collapsed TreeSelectField

### PR DESCRIPTION
**Pull request by 0ces**
Allows to filter options on a tree select field even if the parent option is collapsed.

### CODE REVIEW

File: src/inputs/TreeSelectField/TreeSelectField.tsx [hunk 1]
```diff
@@ -325,6 +325,7 @@ function TreeSelectFieldBase<O, V extends Value>(props: TreeSelectFieldProps<O,
   // Update the filtered options when the input value changes
   const onInputChange = useCallback(
     (inputValue: string) => {
+      console.log({ inputValue });
```
Comment (stephenh) [inline src/inputs/TreeSelectField/TreeSelectField.tsx:328]: Nit: snuck in

File: src/inputs/TreeSelectField/TreeSelectField.test.tsx [hunk 2]
```diff
@@ -394,6 +394,32 @@ describe(TreeSelectField, () => {
...
+      />,
+    );
+    // When opening the options
+    click(r.favoriteLeague);
+    // Then the child options are visible
+    expect(r.getByRole("option", { name: "MLB" })).toBeVisible();
+    // When we collapse the parent option
+    click(r.treeOption_collapseToggle_basketball);
+    // And typing in the filter input
+    fireEvent.input(r.favoriteLeague, { target: { value: "nba" } });
```
Comment (stephenh) [inline src/inputs/TreeSelectField/TreeSelectField.test.tsx:416]: Kinda wonder why this isn't using `type`... :shrug:

**Review by stephenh — APPROVED** [review]
Makes sense!

File: src/inputs/TreeSelectField/TreeSelectField.test.tsx (hunk 2 above)
Comment (0ces) [reply src/inputs/TreeSelectField/TreeSelectField.test.tsx:416]: I just copied from one of the tests above, using `type` doesn't work :/

**Pull request merged**